content/guest-post/loopback-rest-api-authentication/index.md

```diff
@@ -0,0 +1,577 @@
+
+- A Linux machine – This tutorial will use Ubuntu 20.04.3 LTS (Focal Fossa). The tutorial also works well on other Linux distributions and operating systems.
+- [**NodeJS**](https://nodejs.org/) – JavaScript runtime built on Chrome's V8 JavaScript engine.
+- [**MongoDB**](https://www.mongodb.com/) — Document-oriented database program.
+
+## Install LoopBack CLI
+
+To start building your LoopBack REST API, first install LoopBack CLI, which provides the quickest method to get started with a LoopBack 4 project that follows best practices.
+
+Use the command below to install the Loopback CLI globally:
+
+```bash
+    npm i -g @loopback/cli
+```
+
+You can grab a cup of coffee while you wait for the installation to complete. Then open your command line, create an `AuthWithLooback` folder, and change directory to the `AuthWithLooback` folder with commands below:
+
+```bash
+    mkdir AuthWithLooback
+    cd AuthWithLooback
+```
+
+## Scaffold Your LoopBack Project
+
+So, you've installed Loopback CLI and created a project directory. Let's run the following command to create a LoopBack project:
+
+```bash
+    lb4 app
+```
+
+Select the options as in the following screenshot to complete the prompts.
+
+![Creating a loopback project](l5wg0nTQ.png)
+
+After completing the prompts, LoopBack will configure the TypeScript compiler and install all the required dependencies. Change directory into the `auth-with-loopback` folder.
+
+```bash
+  cd auth-with-loopback
+```
+
+## Create Model
+
+You've successfully created your Loopback application. Now, let’s create a Model to store the news details with the command below:
+
+```bash
+  lb4 model
+```
+
+Select the options as in the following screenshot to complete the prompts.
+
+![Creating a News Model](PFv-ZD0w.png)
+
+After the `date_created` property definition, press the enter key to exit the prompt.
+
+Loopback will create a `NewsModel` file in the `src/models` — the folder where `NewsModel` will be defined.
+
+Next, you need to create a data source to connect to your preferred database. For demonstration, this tutorial connects to a MongoDB database.
+
+Run the follwing command in your terminal to create a data source:
+
+```bash
+  lb4 datasource
+```
+
+Select the options as in the following screenshot to complete the prompts.
+
+![Creating News datasource](pHvPgEQA.png)
+
+After completing the prompts, LoopBack will create the `News` file in the `src/datasource` folder.
+
+Then, create a [Repository](https://loopback.io/doc/en/lb4/Repository.html) for CRUD operations of your NewModel with the command below:
+
+```bash
+  lb4 repository
+```
+
+After completing the prompts, LoopBack will create the `NewsModelRepository` file in the `src/repository` folder.
+
+Select `NewsDatasource` as the data source, `NewsModel` as the to generate repository for, and `DefaultCrudRepository` as the base repository class.
```

Comment:
@icode247 This one phrase is little bit unclear: 

```
`NewsModel` as the to generate repository for
```
`as the to` is ambiguous here

Reply:
Facing following error after running command `npm start` which is run after `lb4 controller`  command.

```

> auth-with-loopback@0.0.1 prestart
> npm run rebuild


> auth-with-loopback@0.0.1 rebuild
> npm run clean && npm run build


> auth-with-loopback@0.0.1 clean
> lb-clean dist *.tsbuildinfo .eslintcache


> auth-with-loopback@0.0.1 build
> lb-tsc

src/repositories/news-model.repository.ts:7:3 - error TS2344: Type 'NewsModel' does not satisfy the constraint 'Entity'.
  Type 'NewsModel' is missing the following properties from type 'Entity': getId, getIdObject 

7   NewsModel,
    ~~~~~~~~~

src/repositories/news-model.repository.ts:14:11 - error TS2345: Argument of type 'typeof NewsModel' is not assignable to parameter of type 'typeof Entity & { prototype: NewsModel; }'.
  Type 'typeof NewsModel' is missing the following properties from type 'typeof Entity': getIdProperties, getIdOf, buildWhereForId

14     super(NewsModel, dataSource);
             ~~~~~~~~~
```

Please help understand and how to resolve this error